jetty-core/jetty-http/src/main/java/org/eclipse/jetty/http/HttpGenerator.java

```diff
@@ -627,23 +628,36 @@ else if (contentLength != field.getLongValue())
 
                         case CONNECTION:
                         {
-                            boolean keepAlive = field.contains(HttpHeaderValue.KEEP_ALIVE.asString());
-                            if (keepAlive && _info.getHttpVersion() == HttpVersion.HTTP_1_0 && _persistent == null)
+                            boolean hasValue = StringUtil.isNotBlank(field.getValue());
+                            boolean hasKeepAlive = field.contains(HttpHeaderValue.KEEP_ALIVE.asString());
+                            boolean hasClose = field.contains(HttpHeaderValue.CLOSE.asString());
+
+                            if (http10 && hasKeepAlive)
                             {
-                                _persistent = true;
+                                // set persistence if unset
+                                if (_persistent == null)
+                                    _persistent = true;
                             }
-                            if (field.contains(HttpHeaderValue.CLOSE.asString()))
+                            else if (hasClose)
                             {
                                 close = true;
                                 _persistent = false;
                             }
-                            if (keepAlive && _persistent == Boolean.FALSE)
+
+                            if (_persistent == Boolean.FALSE && hasKeepAlive)
                             {
-                                field = new HttpField(HttpHeader.CONNECTION,
-                                    Stream.of(field.getValues()).filter(s -> !HttpHeaderValue.KEEP_ALIVE.is(s))
-                                        .collect(Collectors.joining(", ")));
+                                // we need to strip the keep-alive values
+                                String resultingValue = Stream.of(field.getValues())
+                                    .filter(s -> !HttpHeaderValue.KEEP_ALIVE.is(s))
+                                    .collect(Collectors.joining(", "));
```

Comment:
I wonder if string operations here would be better than streams...
Also perhaps we could do this in the condition bodies above?

Reply:
Yeah, I'm not a fan of the stream operation here, but it was here before I started this, so I left it.
I was going to look into adding a `HttpField.removeValue(String)` method as a possible replacement.

Reply:
Yeah, so that.   I agree a removeValue method would allow us to hide the details.